File: matrixSearch.java
import java.util.Scanner;

public class matrixSearch {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int row = sc.nextInt();
        int colum = sc.nextInt();
        int [][] matrix = new int[row][colum];
        for(int i = 0; i < row; i++){
            for(int j = 0; j < colum; j++){
                matrix[i][j] = sc.nextInt();
            }
        }
        int check = sc.nextInt();
        int flag = 0;
        for(int i = 0; i < row;i++){
            for(int j = 0; j < colum; j++){
                if(matrix[i][j] == check){
                    flag = 1;
                    break;
                }
            }
            if(flag == 1){
                System.out.println("will not take number");
                break;
            }
        }
        if(flag != 1){
            System.out.println("will take number");
        }
    }
}

File: matrixSearch.py
row = int(input())
colum = int(input())
Matrix = [[0 for _ in range(row)] for _ in range(colum)]
for i in range(row):
    for j in range(colum):
        Matrix[i][j] = int(input())
check = int(input())
flag = 0
for z in range(row):
    for k in range(colum):
        if Matrix[z][k] == check:
            flag = 1
            break
if flag == 1:
    print('will not take number') 
else:
    print('will take number')       

